datastore-cli/src/main/scala/org/allenai/datastore/cli/UploadApp.scala

```diff
@@ -42,20 +42,22 @@ object UploadApp extends App {
     help("help")
   }
 
-  parser.parse(args, Config()) foreach { config =>
-    val datastore = config.datastore match {
-      case Some(datastore) => datastore
-      case None =>
-        Common.printDefaultDatastoreWarning()
-        Datastore
+  Common.handleDatastoreExceptions {
+    parser.parse(args, Config()) foreach { config =>
+      val datastore = config.datastore match {
```

Comment:
`getOrElse`? Or pull into a helper on `Config`?

Also, you should really rename `Config` - it's confusing when it's not a typesafe config object.